packages/smooth_app/lib/data_models/continuous_scan_model.dart

```diff
@@ -8,62 +8,65 @@ import 'package:smooth_app/cards/product_cards/smooth_product_card_found.dart';
 import 'package:smooth_app/cards/product_cards/smooth_product_card_not_found.dart';
 import 'package:smooth_app/cards/product_cards/smooth_product_card_loading.dart';
 import 'package:smooth_app/cards/product_cards/smooth_product_card_thanks.dart';
-import 'package:smooth_app/database/full_products_database.dart';
+import 'package:smooth_app/database/barcode_product_query.dart';
+import 'package:smooth_app/database/local_database.dart';
 
 enum ScannedProductState { FOUND, NOT_FOUND, LOADING }
 
-class ContinuousScanModel extends ChangeNotifier {
```

Comment:
Is there a valid reason to get rid of the `ChangeNotifier` ?
We are using Provider as State Management and the `StatefulWidgets` should be avoided (Now that I think about it I shouldn't have validated the new Profile page as it doesn't follow this guideline).

Provider works by separating the UI from the logic, every Widget can be stateless and the class that extends `ChangeNotifier` holds all the variables, calling notifyListener() tells the Widget to rebuild.

Sorry that this wasn't specified anywhere on the project, I will update the guidelines.

Reply:
I understand your point about the `ChangeNotifier` vs. `StatefulWidget`.
That being said, the code before I refactored it was really painful to read, with `Consumer`s here and there.
My idea is to have most of the relevant data in the `Provider`s that are already in the `main.dart` - in that case we could notify each time the scan list is updated as a product list stored in the database, for instance.
Interesting point, to be investigated.
Is there a link for the guidelines?

Reply:
I agree that the use of `Consumers` is quite heavy but I also fear that mixing `ChangeNotifiers` and `StatefulWidgets` can become confusing. Perhaps we should determine a clear rule on when `Provider` or `StatefulWidgets` should be used.
Having the data at a higher levels such as `main.dart` sounds good to me, I don't really see how that will reduce the use of `Consumers` though.

There is actually no guidelines at the moment, only the technical document you already saw for the choice of database. I will start working on guidelines and documentation for future contributors.

Reply:
> I don't really see how that will reduce the use of Consumers though.

That's the beauty of `Provider`s: if I start my `build(BuildContext context)` with a `context.watch<LocalDatabase>()`, every time the notify listener method is called on `LocalDatabase`, the `Widget` is rebuilt. And I don't need any `Consumer`.

Don't put too much effort in the guidelines for the moment; I think we're still in an early stage, with many open questions. I guess a simple `README` would do the job in first approach.

Reply:
Oh indeed that's really clever!
Should we split this PR by keeping only the database part, or do you think you can update it to support this new data structure?

I will take some note and make a small file with the essential guidelines

Reply:
I did not want to change the screens in the first place, but I had too, because of the impact of the change of database.
If you give me a couple of days I think I can do something nice with `ContinuousScanPage`, in this PR. That will give us food for thought, and perhaps ideas of templates / for your guidelines.

Reply:
I understand, that would be great. Once we have this page working we can transition the rest of the app to use the same principle and the document it (or probably the other way around).
I will wait for your update and we can discuss it then, let me know if you need any help.